### ISSUE TITLE
New Model SLEEPFM

### ISSUE THREAD

**Issue by FazlaMashrur** [braindecode/braindecode#925]
The only mode from Stanford that is dedicated to sleep research, published in Nature Medicine. I request that you add this to the braindecode. 

Publication: https://www.nature.com/articles/s41591-025-04133-4

GitHub: https://github.com/zou-group/sleepfm-clinical

**Comment by bruAristimunha** [comment]
We are open to contributions @FazlaMashrur. I think at the moment, nobody of the maintainer team is working with sleep data, i can review your contribution!

**Comment by bruAristimunha** [comment]
Act @Fashad-Ahmed, I think this model is better to be implemented! if you can list the baseline that he used, it would be great!

**Comment by Fashad-Ahmed** [comment]
> Act [@Fashad-Ahmed](https://github.com/Fashad-Ahmed), I think this model is better to be implemented! if you can list the baseline that he used, it would be great!

Sure, @bruAristimunha  I will go through it and update in the thread soon. Thanks for mentioning

**Comment by Fashad-Ahmed** [comment]
@bruAristimunha apologies for implementing it late; I was having my exams, so I was quite occupied there. Please review it and let me know your feedback thanks.


These could the potential follow up PRs:

  - Pretrained-weight download and caching API.
  - SleepFM-specific PSG preprocessing and channel-mapping utilities.
  - End-to-end sleep-staging and fine-tuning example.
  - General support for token-wise/sequence model outputs.
  - Variable-montage batching and channel-mask utilities.
  - Disease-prediction heads with demographic fusion.
  - Cox proportional-hazards loss, metrics, and training support.
  - Demographics and End-to-End PSG baseline implementations.
  - Benchmark reproduction using consistent splits and metrics.
  - Extended numerical-parity tests against the upstream implementation.

**Comment by bruAristimunha** [comment]
Hey @Fashad-Ahmed, don't worry! I will depth review the code and many thanks!
